internal/dependency/kv_get.go

```diff
@@ -17,47 +17,24 @@ var (
 	KVGetQueryRe = regexp.MustCompile(`\A` + keyRe + dcRe + `\z`)
 )
 
-// KVExistsQuery uses a non-blocking query with the KV store for key lookup.
-type KVExistsQuery struct {
-	isConsul
-	stopCh chan struct{}
-
-	dc   string
-	key  string
-	opts QueryOptions
-}
-
 // KVGetQuery queries the KV store for a single key.
 type KVGetQuery struct {
 	KVExistsQuery
 	isBlocking
 }
 
-func (d *KVExistsQuery) SetOptions(opts QueryOptions) {
-	opts.WaitIndex = 0
-	opts.WaitTime = 0
-	d.opts = opts
-}
-func (d *KVExistsQuery) String() string {
-	key := d.key
-	if d.dc != "" {
-		key = key + "@" + d.dc
+// NewKVGetQueryV1 processes options in the format of "key key=value"
+// e.g. "my/key dc=dc1"
+func NewKVGetQueryV1(key string, opts []string) (*KVGetQuery, error) {
+	if key == "" {
```

Comment:
Just wanted to follow up with where I'm at! Another issue I ran into was that the input format is different between the two methods, where one takes a string with the options included and the other takes a string for the key and options as a slice. Still trying to figure out a way to make it work since we'll be running into this code duplication every time we add a new V1 method.